### ISSUE TITLE
v5 - Preload fails on HTML slides (because "lazyLoadSlide" returns undefined)

### ISSUE THREAD

**Issue by Stanko** [dimsemenov/PhotoSwipe#1860]
## Context

I'm using Photoswipe to display videos as well by adding video thumbs to the gallery:
```html
<a
  href="/path/to/video.mp4"
  target="_blank"
  data-pswp-width="1600"
  data-pswp-height="900"
  data-is-video="true"
>
  <img src="/video/thumb.jpg" alt="" />
</a>
```

Then, I use `itemData` event to replace `src` with `html`:
```js
lightbox.on("itemData", (e) => {
  if (
    e.itemData &&
    e.itemData.element &&
    e.itemData.element.dataset.isVideo
  ) {
    e.itemData.html = `<div><video src="${e.itemData.src}" controls /></div>`;
    e.itemData.src = undefined;
  }
});
```

But then preload fails, when it gets to the video slide as it doesn't have preloadable `src`:
<img width="569" alt="Screen Shot 2022-03-08 at 12 54 31 PM" src="https://user-images.githubusercontent.com/776788/157233467-354be330-5e4e-457d-b096-907690c65fef.png">

## Potential fix

It seems that the problem happens because `lazyLoadSlide` can return `undefined` but `loadSlideByIndex` method 
assumes it will return an object.

I think this code block:
https://github.com/dimsemenov/PhotoSwipe/blob/v5-beta/src/js/slide/loader.js#L125-L129

should be updated to:
```js
    if (!content) {
      content = lazyLoadSlide(index, this.pswp);

      // lazyLoadSlide can return "undefined", check before adding to cache
      if (content) {
        content.key = key;
        this.addToCache(content);
      }
    }
```

If you think this is appropriate fix, I will gladly open a PR.

### Thank you

This library is fantastic! Cheers!

### EDIT
Issue might be related to https://github.com/dimsemenov/PhotoSwipe/issues/1855

**Comment by end2endi** [comment]
This update resolves the issue I mentioned in #1855

**Comment by dimsemenov** [comment]
Should be fixed by the latest build (5.2.0-beta.2). The beta branch is also on npm now.

**Comment by Stanko** [comment]
Thank you!
